### PR TITLE
rename package to ClimaCore.jl

### LINKED ISSUE #64: Rename to ClimaCore.jl
ClimateMachineCore.jl was a temporary name. The software call agreed to go with ClimaCore.jl.

### PR DESCRIPTION
**Pull request by jakebolewski**
Closes #64

### CODE REVIEW

File: bors.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@ status = [
   "test-os (ubuntu-latest)",
   "test-os (windows-latest)",
   "test-os (macos-latest)",
-  "buildkite/climatemachinecore-ci",
+  "buildkite/ClimaCore-ci",
```
Comment (charleskawczynski) [inline bors.toml:5]:
```suggestion
  "buildkite/climacore-ci",
```
I think?

Comment (jakebolewski) [reply bors.toml:5]: it does seem like it is case sensitive

**Pull request merged**